src/app/AttributeAccessInterface.h

```diff
@@ -126,6 +126,27 @@ class AttributeAccessInterface
      */
     virtual CHIP_ERROR Read(const ConcreteAttributePath & aPath, AttributeValueEncoder & aEncoder) = 0;
 
+    /**
+     * Callback for writing attributes.
+     *
+     * @param [in] aPath indicates which exact data is being write.
+     * @param [in] aTLVReader  A pointer to a TLVReader, which should point to the beginning
+     *                         of this AttributeDataElement to write.
+     * @param [out] aDataWrite whether we actually tried to write data.  If
+     *                         this function returns success and aDataWrite is
+     *                         false, the AttributeAccessInterface did not try
+     *                         to write any data.  In this case, normal attribute
+     *                         access will happen for the write.  This may involve
+     *                         writing to the attribute store or external attribute
+     *                         callbacks.
+     */
+    virtual CHIP_ERROR Write(const ConcreteAttributePath & aPath, TLV::TLVReader & aReader, bool * aDataWrite)
```

Comment:
Whose responsibility is it to write out the status code in the write response? (i.e to call `AddStatus` on the `WriteHandler` object).

Is it the implementer of this function? This needs to be clearly specified in the comment body.

One option includes to return back an `AttributeStatusIB` object that can contain either IM code or cluster-specific status code, and avoid the implementor having to worry about writing out the response.

The other option is to have the implementor manage all error handling (and as a side-effect, make this method return a void). That however, feels less attractive though.

Reply:
It is handled by this logic

    if (dataWrite)
        {
            return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::Success);
        }

Reply:
Per sync, this will be handled in follow-up PR together with read. https://github.com/project-chip/connectedhomeip/issues/11663